### PR TITLE
Rework derelict to provide middleware only

### PR DESCRIPTION
**Pull request by minocys**
### CHANGES
1. Try re-implementing using composition
2. Expose middleware only, remove express router and authenticator as exports

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,53 +1,71 @@
...
+      const jwt = JwtHelpers(config.secret);
+      useXsrf = config.hasOwnProperty(useXsrf) ? config.useXsrf : true;
+      authenticator = Authenticator(jwt, config.createUser, config.fetchUser, useXsrf);
+      augmentRequest(jwt, config.authRules, useXsrf);
+    },
+
+    signUp(req, res) {
+      const newUser = req.body;
+      
+      console.log(newUser);
```
Comment (dtp263) [inline src/index.js:20]: Eww

File: src/passwordHelpers.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  return new Promise((resolve, reject) => {
+    bcrypt.compare(password, user.password, (error, isAuth) => {
+      isAuth ? resolve(isAuth) : reject(error);
+    });
+  });
+}
+
+export function hashPass(password) {
+  return new Promise((resolve, reject) => {
+    bcrypt.hash(password, null, null, (error, hash) => {
```
Comment (dtp263) [inline src/passwordHelpers.js:13]: Why not use a salt? We use the id of a user appended to a random string to make the hashes better

File: src/index.js (hunk 1 above)
Comment (minocys) [inline src/index.js:20]: What's wrong with this?

File: src/passwordHelpers.js (hunk 2 above)
Comment (minocys) [inline src/passwordHelpers.js:13]: haven't implemented salts yet, is a todo

File: src/index.js (hunk 1 above)
Comment (dtp263) [inline src/index.js:20]: You just shouldnt have console output. Unless you want to leave them and then take them out for production. But thats just an extra step later

File: src/Authenticator.js [hunk 3]
```diff
@@ -1,56 +1,59 @@
...
-    this.authRules = authRules;
+export default function Authenticator({ generateJWT }, createUser, fetchUser, useXsrf = true) {
+  
+  function makeTokens(user) {
+    // Generate XSRF
+    const { secret = null, token = null } = useXsrf ? generateXSRF() : {};
 
-    this.augmentRequest();
+    return {
+      user,
```
Comment (dtp263) [inline src/Authenticator.js:11]: this is optional / nitpicky.... But you could give the user object a named index to separate out the user data, and also follow the same pattern as the other ones. 

```
{
  user: user,
  JWT: generateJWT(user, secret),
  XSRF: token
}
```

File: src/passwordHelpers.js (hunk 2 above)
Comment (dtp263) [inline src/passwordHelpers.js:13]: its sooooooooo simple. Just save your random string as a env var then concat that with the user id. and Boom! way more secure of a hash for such little work

**Pull request merged**